internal/util/debug/debug_test.go

```diff
@@ -14,8 +14,83 @@
 
 package debug
 
-import "testing"
+import (
+	"archive/zip"
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"path/filepath"
+	"testing"
+	"time"
 
-func TestDummy(t *testing.T) {
-	// we need at least one test per package to correctly calculate coverage
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"github.com/FerretDB/FerretDB/internal/util/state"
+)
+
+func TestRunHandler(t *testing.T) {
+	t.Parallel()
+
+	host := "127.0.0.1:5454"
+
+	ctx := context.Background()
+
+	filename := filepath.Join(t.TempDir(), "state.json")
+	stateProvider, err := state.NewProvider(filename)
+	require.NoError(t, err)
+
+	metricsRegisterer := prometheus.DefaultRegisterer
+	metricsProvider := stateProvider.MetricsCollector(true)
+	metricsRegisterer.MustRegister(metricsProvider)
+
+	l := zap.S()
+
+	go RunHandler(ctx, host, metricsRegisterer, l.Named("debug").Desugar())
```

Comment:
That goroutine is never stopped

Reply:
It's a test and thread shall stop automatically as soon as test ends. Do we need explicit stop?

Reply:
WDYT?

Reply:
I'd gracefully stop by setting a cancellable context and calling `cancel()` from a `t.Cleanup()` just to avoid any log messages that might confuse me.